File: SDKs/noyo.com/1.0/go/pkg/models/operations/submitapplicationdocument.go
package operations

import (
	"openapi/pkg/models/shared"
)

type SubmitApplicationDocumentPathParams struct {
	ApplicationID string `pathParam:"style=simple,explode=false,name=application_id"`
	DocumentID    string `pathParam:"style=simple,explode=false,name=document_id"`
	Version       string `pathParam:"style=simple,explode=false,name=version"`
}

type SubmitApplicationDocumentRequest struct {
	PathParams SubmitApplicationDocumentPathParams
	Request    shared.ApplicationDocumentEditRequest `request:"mediaType=application/json"`
}

type SubmitApplicationDocumentResponse struct {
	ApplicationDocumentResult *shared.ApplicationDocumentResult
	ContentType               string
	StatusCode                int64
}

File: SDKs/noyo.com/1.0/typescript/src/sdk/models/operations/submitapplicationdocument.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class SubmitApplicationDocumentPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=application_id" })
  applicationId: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=document_id" })
  documentId: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=version" })
  version: string;
}


export class SubmitApplicationDocumentRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: SubmitApplicationDocumentPathParams;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request: shared.ApplicationDocumentEditRequest;
}


export class SubmitApplicationDocumentResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  applicationDocumentResult?: shared.ApplicationDocumentResult;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}
